server/src/group/group.service.ts

```diff
@@ -53,4 +54,47 @@ export class GroupService {
     const groupMember = await user.groupMember!.load();
     return groupMember!.group.load();
   }
+
+  /** Get group from the id.
+   * Throws a NotFoundError if the id does not correspond to a group. */
+  async getGroupFromID(id: string): Promise<Group> {
+    const group = await this.groupsRepository.findOne({ friendlyId: id });
+    if (group == null) throw NotFoundError;
+    return group;
+  }
+
+  /** Helper function that removes a user from their current group.
+   * If the user is the only member, deletes the group. */
+  async removeUserFromGroup(user: User) {
```

Comment:
Consider returning Promise<Group | null> where it returns group if it still exists and null if it does not. It'll help you when you are making members of the old group (if the group exists that is) aware that someone has left.